ksqldb-engine/src/main/java/io/confluent/ksql/ddl/commands/CreateSourceFactory.java

```diff
@@ -200,6 +205,22 @@ public CreateTableCommand createTableCommand(
     );
   }
 
+  private void throwIfCreateOrReplaceOnSourceStreamOrTable(
+      final CreateSource createSource,
+      final DataSource existingSource
+  ) {
+    if (createSource.isOrReplace()) {
+      final String createSourceType = (createSource instanceof CreateStream) ? "stream" : "table";
+
+      if (createSource.isSource() || (existingSource != null && existingSource.isSource())) {
+        throw new KsqlException(
+            String.format("Cannot add %s '%s': CREATE OR REPLACE is not supported on "
+                    + "source streams or tables.",
```

Comment:
nit: `"source streams or tables."` -> change to  `source stream` for CS and `source table` for CT statement -- we have the information already at hand via `createSourceType`

Reply:
Done